app/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailVerificationService.scala

```diff
@@ -65,38 +86,36 @@ class EmailVerificationService @Inject() (
     request: AuthenticatedEnrolledRequest[AnyContent],
     ec: ExecutionContext,
     hc: HeaderCarrier
-  ): Future[Result] = for {
-    verificationId <- addVerificationRequest(request.eoriNumber, email)
-    verificationResponse <- verifyEmail(nextPageUrl(verificationId), previousPage.url)(request.authorityId, email)
-  } yield verificationResponse.fold(Redirect(previousPage))(value =>
-    Redirect(generateEmailServiceUrl(value.redirectUri))
-  )
+  ): Future[Result] = {
+    val credId = request.authorityId
+    for {
+      verificationId <- addVerificationRequest(request.eoriNumber, email)
+      verificationResponse <- verifyEmail(
+        verifyEmailUrl = nextPageUrl(verificationId),
+        confirmEmailUrl = previousPage.url,
+        credId = credId,
+        email = email
+      )
+    } yield verificationResponse.fold {
+      Redirect(previousPage)
+    }(value => Redirect(generateEmailServiceUrl(value.redirectUri)))
+  }
 
-  private def verifyEmail(
-    verifyEmailUrl: String,
-    confirmEmailUrl: String
-  )(credId: String, email: String)(implicit
+  private def verifyEmail(verifyEmailUrl: String, confirmEmailUrl: String, credId: String, email: String)(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext,
     request: AuthenticatedEnrolledRequest[AnyContent]
-  ): Future[Option[EmailVerificationResponse]] =
+  ): Future[Option[EmailVerificationResponse]] = {
+    val redirectVerifyEmailUrl = request.toRedirectTarget(verifyEmailUrl)
+    val backUrl = request.toRedirectTarget(confirmEmailUrl)
+
     emailVerificationConnector
       .verifyEmail(
-        EmailVerificationRequest(
+        EmailVerificationRequest.createVerifyEmailRequest(
```

Comment:
Moved to a factory method so we can test that method easily. What this was doing was not tested as the mock it was passed to was just wildcarding everything :(

Wildcarding mocks is very sieve like. It is saying I do not care what you pass me I am always going to return the same answer. A good way to test the quality is see if things fail if you actually change the values passed in the code.

What happens with too much mocking is that the acceptance tests are actually doing the trustworthy testing.